Docs 11618 cpu affinity v4.4 cv (#2129)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/runtime-installation-task.adoc

> = Download, Install, Configure, Upgrade Mule

Reviewer: ```suggestion
= Download, Install, Configure, and Upgrade Mule
```

The procedure for making the `MULE_HOME` setting permanent depends on your operating system.

=== Test Your Mule Instance

> After downloading Mule, you can test if it runs in your system without errors by executing the following commands:

Reviewer: ```suggestion
After downloading Mule, test to confirm that it runs in your system without errors by executing the following commands:
```

To stop Mule, press `CTRL-C` in the terminal in which the script is running.

== Install Mule

> You can install Mule as a Windows Service or a Linux/Unix Daemon by executing the corresponding command:

Reviewer: ```suggestion
You can install Mule as a Windows service or as a Linux or Unix daemon by executing the corresponding command:
```

> * On Windows environments:

Reviewer: ```suggestion
* In Windows environments:
```

+
[source,powershell]
----
$ $MULE_HOME\bin\mule.bat install
----

> * On Linux/Unix environments:

Reviewer: ```suggestion
* In Linux or Unix environments:
```

+
[source,console]
----
$ $MULE_HOME/bin/mule install
----

== Configure Properties in the wrapper.conf File

> After the installation of Mule as a Windows service, you can override the properties defined in the `wrapper.conf` file by declaring new properties in the `wrapper.additional.conf` file.

Reviewer: ```suggestion
After installing Mule as a Windows service, you can override the properties defined in the `wrapper.conf` file by declaring new properties in the `wrapper.additional.conf` file.
```

> The behavior changes depending on the Mule edition you are using:

Reviewer: What do we mean by "the behavior"? How Mule responds to property override? Clarify.

* Mule Community Edition
+

> . After adding a new property, reinstall or restart the Windows Service.

Reviewer: ```suggestion
. After adding a new property, reinstall or restart the Windows service.
```
I ran an online search and it appears that "service" is not part of a proper noun Windows Service.

> . Add the property and ensure that its index is greater than the greatest index in the `wrapper-additional.conf` file. Otherwise, your property can be overridden.

Reviewer: Is "greater" the most precise modifier for "index" ("index value"?)? 
"larger"? 
"higher"? 
Something else? If so, STET

Reviewer: Best guess suggestion:
```suggestion
. Add the property, ensuring that its index is greater than the greatest index in the `wrapper-additional.conf` file so that your property cannot be overridden.
```

+
For example, if the `wrapper-additional.conf` has the property:
+
`wrapper.java.additional.n=-Dproperty=value`
+
When `n` is the greatest index, you should add your property in the `wrapper.conf` file incrementing the index:
+

> `wrapper.java.additional.n+1=-DyourProperty=valueOfYourProperty`

Reviewer: Avoid breaking sentences around other display values:
```suggestion
For example, imagine that `wrapper-additional.conf` has the following property:
+
`wrapper.java.additional.n=-Dproperty=value`
+
If `n` is the greatest index, you would add your property in the `wrapper.conf` file and increment the index, as follows:
+
`wrapper.java.additional.n+1=-DyourProperty=valueOfYourProperty`
```

* Mule Enterprise Edition
+

> . By default, you can add  20 extra properties after installing Mule as a Service.

Reviewer: ```suggestion
. By default, you can add 20 more properties after installing Mule as a service.
```
...but this is not really a step. Consider omitting this altogether and replacing it with....

> . Configure the number of extra properties you need, using the following command line argument

Reviewer: ```suggestion
Configure up to 20 additional properties, using the following command-line argument when you install the service:
```

> when you install the service:

Reviewer: ```suggestion
```

+
`-additionalJavaProperties=numberOfExtraProperties`

== Configure CPU Affinity

> Many operating systems provide a mechanism to bind a process to one or more central processing units (CPUs) so that the process executes only on the assigned CPUs. You can configure CPU affinity in your environment to run Mule in a non-virtualized multi-core server and comply with the number of cores entitled by your subscription.

Reviewer: We generally avoid hyphenating after prefixes, per style ref.:
```suggestion
Many operating systems provide a mechanism to bind a process to one or more central processing units (CPUs) so that the process executes only on the assigned CPUs. You can configure CPU affinity in your environment to run Mule in a nonvirtualized multicore server and comply with the number of cores included in your subscription.
```

=== Considerations When Using CPU Affinity

> * CPUs have groups of cores that share a cache, where thread communication and migration has better performance. Check with your infrastructure team which cores share this cache before setting up this feature to avoid a negative impact to performance.

Reviewer: ```suggestion
* CPUs have groups of cores that share a cache to improve thread communication and migration performance. To avoid such performance issues, check with your infrastructure team to identify which cores share cache before setting up this feature.
```

> * Check with your infrastructure team if there is any resource allocation policy that can interfere with your CPU affinity configuration.

Reviewer: ```suggestion
* Check with your infrastructure team to ensure that no existing resource allocation policy can interfere with your CPU affinity configuration.
```

> * Ensure that the number of cores you configure through CPU affinity is in compliance with the number of cores entitlted by your MuleSoft subscription plan.

Reviewer: ```suggestion
* Ensure that the number of cores you plan to configure through CPU affinity does not exceed the number of cores included in your MuleSoft subscription plan.
```

== Remove the Mule Service or Daemon

Run the `remove` command to remove the service or daemon from your system.

The following example removes the Mule daemon from a Unix console:

[source,console,linenums]
----
$ $MULE_HOME/bin/mule remove
MULE_HOME is set to /Applications/mule-enterprise-standalone-4.3.0
MULE_BASE is set to /Applications/mule-enterprise-standalone-4.3.0
Detected Mac OSX:
Mule Enterprise Edition is not running.
Removing Mule Enterprise Edition daemon...
----

== Install Anypoint Monitoring

> You can xref:monitoring::am-installing.adoc[install Anypoint Monitoring] for cloud-managed supported versions of on-premises runtimes so you can monitor applications running on that server.

Reviewer: ```suggestion
You can xref:monitoring::am-installing.adoc[install Anypoint Monitoring] for cloud-managed supported versions of on-premises runtime instances to monitor applications running on that server.
```
